### ISSUE TITLE
Instructions for manpage installation

### ISSUE THREAD

**Issue by karel-brinda** [bioconda/bioconda-recipes#4121]
I think that it would be good to write a paragraph about man pages installation in conda recipes. Man pages of some programs are not being installed at all (e.g., for miniasm), or they are installed incorrectly (e.g., for ococo – see https://github.com/karel-brinda/ococo/issues/23, I don't know what is the nature of this problem).

Since manpages are one of the main information sources on Unix, there also might be a policy that bioconda recipes should install all the associated man pages..

**Comment by johanneskoester** [comment]
Indeed. This is a good idea. As far as I know, Miniconda correctly sets the man page search path (see output of `manpath`). We just need to ensure that the pages are installed to that path.

**Comment by karel-brinda** [comment]
I think that it would be good to provide examples of how to do it correctly. For instance, 
```
MANDIR    = $(PREFIX)/share/man/man1
MANPAGE   = ococo.1
install  $(MANPAGE) $(MANDIR)/$(MANPAGE)
```
is apparently not the way how it should be done.

**Comment by acaprez** [comment]
I don't think `conda` does anything with the man search path directly.  A little searching found this: conda/conda#845.  Basically, the `man` command itself tries to be smart and searches directories "near" the binary.  The main issue is that what it considers "near" seems to vary from platform to platform.  In my F25 VM, `manpath` returns `${PREFIX}/man` but not `${PREFIX}/share/man`.  However, in my OS X VM it's the opposite; it returns `${PREFIX}/share/man` but not `${PREFIX}/man`.  So to ensure it worked everywhere, you'd need to create `${PREFIX}/share/man` and symlink to `${PREFIX}/man` (or vice-versa) I believe (assuming all other platforms searched one or the other).  Or alternatively, add an `activate.d` script to set `MANPATH` uniformly everywhere.

**Comment by johanneskoester** [comment]
@acaprez, are you sure? This is what I get:
```
$ manpath
/home/johannes/.local/opt/miniconda/man:/usr/local/share/man:/usr/share/man
```
Hence, conda seems to specifically set the manpath, right? Therefore, we would just need to ensure that man pages are installed to `$PREFIX/man` instead or `$PREFIX/share/man`, right?

**Comment by johanneskoester** [comment]
@acaprez and @karel-brinda I have added MANPATH handling to activate and deactivate scripts here in my fork of conda:
https://github.com/johanneskoester/conda/tree/manpath
(see shell/activate.sh and shell/deactivate.sh)
Could you test and give me your opinion? If you think this is what we need, please let me know and I will create a pull request to upstream.

**Comment by karel-brinda** [comment]
What is the correct way of copying manpages in Conda? I had certain problems with Ococo, but I can't reproduce them now. When I installed ococo using `conda install ococo`, the installation process worked. When I wanted to create a new environment and I included ococo in the environment specification, it broke the entire Conda (the error message is here: https://github.com/karel-brinda/ococo/issues/23).

**Comment by karel-brinda** [comment]
@johanneskoester I tested your script and I think that I understand several of the problems now. It looks like that Conda has changed the way of handling manpages since the last time I tested it. 

First, adding `/Users/karel/miniconda/share/man` to `$MANPATH` to be useless for me because Conda does not put manpages into this path any more (but it used to be there). For instance, the manpage for ococo is now in `/Users/karel/miniconda/pkgs/ococo-0.1.2.4-0/share/man`

I think that your activation script should put a slightly different path to `$MANPATH`. Now it is `~/miniconda/envs/$envname/man`, but I think that it should be `~/miniconda/envs/$envname/share/man`.

Finally, Ococo manpage appears to be in `/Users/karel/miniconda/envs/$envname/share/man/man1` instead of `/Users/karel/miniconda/envs/$envname/share/man/man/ococo.1`. I think that this is a consequence of this command: `https://github.com/bioconda/bioconda-recipes/blob/97b21e9871443c159444474a0e5d08267e674dd6/recipes/ococo/build.sh#L28` so the directory `man1` probably does not exist at time of copying. This is apparently the root of my old problem reported in `https://github.com/karel-brinda/ococo/issues/23`. So the recommendation for authors of recipes is that they should ensure existence of the man directory using `mkdir -p` before copying the manpages.

**Comment by johanneskoester** [comment]
@karel-brinda I have updated the fork to also add `share/man`. However, there is now a problem when doing `source deactivate`: assume I am in environment A, and activate environment B. Now, deactivating B will remove more than only the manpaths of B from `$MANPATH`. I am not a real bash nerd, so I would appreciate any advice.
